core/mondoo-ms365-security.mql.yaml

```diff
@@ -0,0 +1,611 @@
+       ### Microsoft 365 Console
+
+        To update via the Microsoft 365 portal:
+        1. Log in as Global Administrator to the Microsoft 365 portal at https://admin.microsoft.com
+        2. Settings --> Org Settings --> Security & Privacy --> Password expiration policy -->  Set passwords to never expire (recommended)
+        3. Select Save 
+    query: |
+      microsoft.domains {
+        passwordValidityPeriodInDays == 2147483647
+      }
+  - uid: mondoo-m365-security-ensure-multifactor-authentication-is-enabled-for-all-users-in-administrative-roles
+    title: Ensure all users in administrative roles have MFA enabled 
+    severity: 100
+    docs:
+      desc: |
+        Enable multifactor authentication for all users in administrative roles!
+      audit: |
+       __cnquery run__
+
+        To audit Microsoft 365 with `cnquery run`:
+
+        Run the following query:
+
+          ```mql
+          cnquery run ms365 -c "microsoft.security.latestSecureScores.controlScores.one( _['controlName'] == 'AdminMFAV2' && _['score'] == 10)" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+
+        __cnquery shell__
+
+        To audit Microsoft 365 with `cnquery shell`:
+
+        1. Launch `cnquery shell`:
+
+          ```bash
+          cnquery shell ms365 --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        
+        2. Run the following query:
+
+          ```mql
+          microsoft.security.latestSecureScores.controlScores.one( _['controlName'] == 'AdminMFAV2' && _['score'] == 10 )
+          ```
+      remediation: |
+         ### Microsoft 365 Console
+         
+         To update via the Microsoft 365 portal:
+         1. Log in as Global Administrator to the Microsoft 365 portal at https://admin.microsoft.com
+         2. Select "Azure Active Directory" once in the "Admin Centers"
+         3. Applications --> Enterprise applications --> Security --> Conditional Access 
+         4. Make sure there is a Policy with following conditions: 
+           * Grant --> Grant access (Require multifactor authentication)
+           * Users --> Users and groups --> Include --> Select users and groups --> Directory roles --> at minimum select following rules (Billing admin, Conditional Access admin, Exchange admin, Global admin, Helpdesk admin, Security admin, SharePoint admin, and User admin)
+           * Cloud apps or actions --> All cloud apps 
+         5. Make sure that the policy is Enabled  
+    query: |
+      microsoft.security.latestSecureScores.controlScores.one( _['controlName'] == 'AdminMFAV2' && _['score'] == 10 )
+  - uid: mondoo-m365-security-ensure-that-between-two-and-four-global-admins-are-designated
+    title: Ensure that at least three global admins are designated and not more than four
+    severity: 80
+    docs:
+      desc: |
+        This control checks that there are enough Global Admins in a single tenant.
```

Comment:
```suggestion
        This control checks that there are enough Global Admins in a single tenant. 
        When it comes to designating global admins, it's important to consider the size and complexity of the organization, as well as the level of responsibility and authority required for the role. As a general rule, it's a good idea to have at least three global admins to ensure that there is redundancy and coverage in case one admin is unavailable or leaves the organization.

At the same time, having too many global admins can lead to confusion and inefficiency, as multiple people may be making decisions or taking actions without proper coordination. Therefore, it's recommended to keep the number of global admins to no more than four, unless the organization is particularly large or complex and requires more administrators to properly manage its operations.
```

Reply:
Thanks, done!